### PR TITLE
[Modules] Updated Network/VirtualNetworkGateways to new dependency approach

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Updated Network/VirtualNetworkGateways to new dependency approach

| Pipeline |
| - |
| [Network: VirtualNetworkGateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualnetworkgateways.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/virtualNetworkGateways/.test/expressRoute/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,88 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/expressRoute/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,86 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: .github/workflows/ms.network.virtualnetworkgateways.yml [hunk 3]
```diff
@@ -106,8 +106,7 @@ jobs:
       - name: 'Using test file [${{ matrix.moduleTestFilePaths }}]'
         uses: ./.github/actions/templates/validateModuleDeployment
         with:
-          templateFilePath: '${{ env.modulePath }}/deploy.bicep'
-          parameterFilePath: '${{ env.modulePath }}/${{ matrix.moduleTestFilePaths }}'
+          templateFilePath: '${{ env.modulePath }}/${{ matrix.moduleTestFilePaths }}'
```
Comment (eriqua) [inline .github/workflows/ms.network.virtualnetworkgateways.yml:109]: Just a comment to ask: should #1844  be considered first?

Comment (MrMCake) [reply .github/workflows/ms.network.virtualnetworkgateways.yml:109]: If it's pipeline would not be red, sure 😄

Comment (MrMCake) [reply .github/workflows/ms.network.virtualnetworkgateways.yml:109]: It appears only the removal failed there. Will re-run it. If the pipeline turns out green I'd vote to merge that first

Comment (MrMCake) [reply .github/workflows/ms.network.virtualnetworkgateways.yml:109]: Converted this PR to draft until the other is done, conflicts are resolved & the pipeline was re-run.

Comment (AlexanderSehr) [reply .github/workflows/ms.network.virtualnetworkgateways.yml:109]: Incorporated the changes. The PR is again ready for review

File: modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    virtualNetworkGatewaySku: 'VpnGw1AZ'
+    virtualNetworkGatewayType: 'Vpn'
+    vNetResourceId: resourceGroupResources.outputs.vnetResourceId
+    // Non-required parameters
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep:62]: Please remove comment
```suggestion
```

File: modules/Microsoft.Network/virtualNetworkGateways/.test/aadvpn/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    virtualNetworkGatewaySku: 'VpnGw2AZ'
+    virtualNetworkGatewayType: 'Vpn'
+    vNetResourceId: resourceGroupResources.outputs.vnetResourceId
+    activeActive: false
+    diagnosticStorageAccountId: diagnosticDependencies.outputs.storageAccountResourceId
+    diagnosticWorkspaceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+    diagnosticEventHubAuthorizationRuleId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+    diagnosticEventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+    domainNameLabel: [
+      '<<namePrefix>>-gw-vpn-dm-${serviceShort}-001'
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/aadvpn/deploy.test.bicep:68]: I'd suggestthis for consistency with other examples
```suggestion
      '<<namePrefix>>-dm-${serviceShort}'
```

File: modules/Microsoft.Network/virtualNetworkGateways/.test/aadvpn/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      {
+        roleDefinitionIdOrName: 'Reader'
+        principalIds: [
+          resourceGroupResources.outputs.managedIdentityPrincipalId
+        ]
+        principalType: 'ServicePrincipal'
+      }
+    ]
+    vpnClientAadConfiguration: {
+      aadAudience: '41b23e61-6c1e-4545-b367-cd054e0ed4b4'
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/aadvpn/deploy.test.bicep:84]: Sorry I didn't notice this in the previous PR. Which guid is this?

Comment (AlexanderSehr) [reply modules/Microsoft.Network/virtualNetworkGateways/.test/aadvpn/deploy.test.bicep:84]: No idea.

Comment (eriqua) [reply modules/Microsoft.Network/virtualNetworkGateways/.test/aadvpn/deploy.test.bicep:84]: [Found it](https://learn.microsoft.com/en-us/azure/vpn-gateway/openvpn-azure-ad-tenant#enable-authentication)
It's the Application ID of the "Azure VPN" Azure AD Enterprise App for Azure Public. I'd suggest a note
```suggestion
      aadAudience: '41b23e61-6c1e-4545-b367-cd054e0ed4b4' // The Application ID of the "Azure VPN" Azure AD Enterprise App for Azure Public
```

File: modules/Microsoft.Network/virtualNetworkGateways/readme.md [hunk 7]
```diff
@@ -329,30 +329,27 @@ module virtualNetworkGateways './Microsoft.Network/virtualNetworkGateways/deploy
...
+      "value": "<diagnosticStorageAccountId>"
     },
     "diagnosticWorkspaceId": {
-      "value": "/subscriptions/<<subscriptionId>>/resourcegroups/validation-rg/providers/microsoft.operationalinsights/workspaces/adp-<<namePrefix>>-az-law-x-001"
+      "value": "<diagnosticWorkspaceId>"
     },
     "domainNameLabel": {
       "value": [
-        "<<namePrefix>>-az-gw-vpn-dm-001"
+        "<<namePrefix>>-gw-vpn-dm-nvngavpn-001"
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualNetworkGateways/readme.md:352]: Should be updated after change

Comment (AlexanderSehr) [reply modules/Microsoft.Network/virtualNetworkGateways/readme.md]: updated

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**